ocaml/xapi-idl/lib/debug_info.ml

```diff
@@ -37,11 +61,18 @@ let filter_separator = Astring.String.filter (( <> ) separator)
 let to_string t =
   Option.fold ~none:t.log
     ~some:(fun span ->
-      let traceparent =
-        Tracing.Span.get_context span |> Tracing.SpanContext.to_traceparent
+      let trace_context =
+        let traceparent =
+          span |> Tracing.Span.get_context |> Tracing.SpanContext.to_traceparent
+        in
+        span
+        |> Tracing.Span.get_context
+        |> Tracing.SpanContext.context_of_span_context
+        |> Tracing.TraceContext.with_traceparent (Some traceparent)
+        |> Tracing.TraceContext.to_json_string
```

Comment:
looks like this section is trying to propagate the span's context by:
* keeping the span's tracecontext.baggage
* and reconstructing the span's tracecontext.traceparent using the span's current spancontext.traceparent?

So, it may be beter to make this explicit by creating a new function named something like `Span.get_propagation_context` that returns the tracecontext that can be used with to_json_string, resulting in something easier to understand in the section above like `span |> Span.get_propagation_context |> Tracing.TraceContext.to_json_string`